### DISCUSSION TITLE
Carla Windows, make PythonAPI error, C:\Program' is not recognized as an internal or external command

### DISCUSSION THREAD

**Discussion by furkanunsal42** [carla-simulator/carla#5351] [Q&A]
`C:\Users\unsal\carla>make PythonAPI
-[Setup]: [Batch params]: --boost-toolset msvc-14.2 --all
-[Setup]: Asynchronous jobs:  16
-[Setup]: Boost toolset:      msvc-14.2
-[Setup]: Install directory:  "C:\Users\unsal\carla\Build\"
-[Setup]: Installing zlib...
    -[install_zlib]: [Batch params]: --build-dir "C:\Users\unsal\carla\Build\"
    -[install_zlib]: Retrieving zlib.
    -[install_zlib]: Extracting zlib from "zlib-1.2.12.zip".
    -[install_zlib]: Removing "zlib-1.2.12.zip"
    -[install_zlib]: Creating "C:\Users\unsal\carla\Build\zlib-source\build"
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- The C compiler identification is MSVC 19.28.29914.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe - broken
CMake Error at C:/Program Files/CMake/share/cmake-3.23/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/unsal/carla/Build/zlib-source/build/CMakeFiles/CMakeTmp

    Run Build Command(s):nmake -f Makefile /nologo cmTC_e3fc7\fast &&   C:\Program Files(x86)\GnuWin32\bin\make.exe  -f CMakeFiles\cmTC_e3fc7.dir\build.make /nologo -L                  CMakeFiles\cmTC_e3fc7.dir\build
    'C:\Program' is not recognized as an internal or external command,
    operable program or batch file.
    NMAKE : fatal error U1077: 'C:\Program' : D÷n³■ kodu '0x1'
    Stop.

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)

-- Configuring incomplete, errors occurred!
See also "C:/Users/unsal/carla/Build/zlib-source/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/unsal/carla/Build/zlib-source/build/CMakeFiles/CMakeError.log".

    -[install_zlib]: [CMAKE ERROR] An error ocurred while executing cmake command.
    -[install_zlib]: [CMAKE ERROR] Possible causes:
    -[install_zlib]:                - Make sure "CMake" is installed.
    -[install_zlib]:                - Make sure it is available on your Windows "path".
    -[install_zlib]:                - Make sure you have cmake 3.12.4 or higher installed.
    -[install_zlib]: Exiting with error...

-[Setup]:
   Ok, and error ocurred, don't panic!
   We have different platform
s where you can find some help :)

   - Make sure you have read the documentation:
       http://carla.readthedocs.io/en/latest/how_to_build_on_windows/

   - If the problem persists, you can ask on our Github's "Building on Windows" issue:
       https://github.com/carla-simulator/carla/issues/21

   - Or just use our Discord channel!
       We'll be glad to help you there :)
       https://discord.gg/42KJdRj
make: *** [Util/BuildTools/Windows.mk:74: setup] Error 1
`

I'm trying to install windows version of Carla, I already have Cmake but when running "make PythonAPI" command I get this error. I suspect the error is caused by the space in directory C:\Program Files(x86)\GnuWin32\bin\make.exe. I thought of using "" in the code for calling the file but couldn't really find a way to do so.

these are error log files in carla/Build/zlib-source/build/CMakeFiles:
[CMakeError.log](https://github.com/carla-simulator/carla/files/8456091/CMakeError.log)
[CMakeOutput.log](https://github.com/carla-simulator/carla/files/8456092/CMakeOutput.log)

**Comment by jorgenad** [comment]
Hello, I had the same error and I had also changed the version of zlib. After much searching, I found on discord the user 'edufrikuto' uploaded the version of zlib that is originally used. I attach the file, you have to delete everything that is inside carla\Build and then put the folders that are inside the zip file. That solved my problem. I hope it helps you
[zlib1_2_11.zip](https://github.com/carla-simulator/carla/files/8568302/zlib1_2_11.zip)
.

**Reply by zoughi** [reply]
Hi,
after a lot of search you solved my problem
thank you

**Reply by leilequshuijiao** [reply]
thanks

**Reply by ronyshaji** [reply]
Hey, did you solved the problem after this ?

**Reply by vinayakap** [reply]
this solved my zlib issue for 0.9.13 version of carla.

**Reply by sexiy** [reply]
god!
